syntax = "proto3";
option cc_enable_arenas = true;

package Ydb.View;
option java_package = "com.yandex.ydb.view";

import "ydb/public/api/protos/annotations/validation.proto";
import "ydb/public/api/protos/ydb_operation.proto";
import "ydb/public/api/protos/ydb_scheme.proto";

message DescribeViewRequest {
    Ydb.Operations.OperationParams operation_params = 1;
    // The path to the view.
    string path = 2 [(required) = true];
}

message DescribeViewResponse {
    // The result of the request will be inside the operation proto.
    Ydb.Operations.Operation operation = 1;
}

message DescribeViewResult {
    // Description of a generic scheme object.
    Ydb.Scheme.Entry self = 1;

    // View-specific fields.
    string query_text = 2;
}
